docs/diagrams/CourseValidator.puml

```diff
@@ -0,0 +1,45 @@
+@startuml
+hide footbox
+!include .SequenceDiagram.config
+participant ":AddCoursesCommand" as AddCoursesCommand
+participant ":CourseValidator" as CourseValidator
+
+activate AddCoursesCommand
+
+AddCoursesCommand -> CourseValidator:isValidInput(nusCourseInput: String, pu: String, puCourseInput: String, jsonObject: JsonObject)
+activate CourseValidator
+
+CourseValidator -> CourseValidator:getPUCourseList
+activate CourseValidator
+
+alt matchPu != null
+CourseValidator -> CourseValidator: get JsonArray for Partner University
+else else
+CourseValidator -> CourseValidator:displayPartnerUniversities()
+activate CourseValidator
+deactivate
+CourseValidator -> CourseValidator: throw IllegalArgumentException
+end
+return courses: JsonArray
+
+
+CourseValidator -> CourseValidator: isValidCourseMapping
+activate CourseValidator
+
+
+loop for each course in courses JsonArray
+CourseValidator -> CourseValidator:getJsonObject(course)
+CourseValidator -> CourseValidator:get puCourseCode from JsonArray
+CourseValidator -> CourseValidator:get nusCourseCode from JsonArray
+
+opt puCourseCode == puCourseCodeInput && nusCourseCode == nusCourseCodeInput
+    CourseValidator --> AddCoursesCommand: true: Boolean
+end
+
+end
+  CourseValidator --> CourseValidator: displayAvailableMappings()
+  CourseValidator --> AddCoursesCommand: false: Boolean
+deactivate
+
```

Comment:
![image](https://github.com/user-attachments/assets/b9dec7f0-0944-42ec-b983-dc443861bd09)
i think print is not a self-invocation, should maybe direct it to `UI`